### PR TITLE
Analyze: exclude useless secrets to reduce execution time

### LINKED ISSUE #45458: Support object label selector for istioctl analyze
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**
Releated to https://github.com/istio/istio/issues/27772

If users use Helm as their application deployment tool, Helm will use secrets as the default storage [doc](https://helm.sh/docs/faq/changes_since_helm2/#secrets-as-the-default-storage-driver)

This will result in a large number of secrets in the cluster, and since it stores chart values, the content is usually very large.

`istioctl analyze` during the initialization phase, it will wait to synchronize all the secrets, but these objects are useless for analysis. If there are too many secrets, it will not only affect the analysis time but may also cause kube-apiserver to fail to synchronize completely.

I want to add a new "resource-filters" parameter so that users can filter out specific objects using a label selector. [doc](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#api) kube-apiserver will not send these objects content to client.

Example:

`istioctl analyze -A -F "Secret owner!=helm"`

In my testing environment, in a scenario with over 2000 secrets, the execution time of `istioctl analyze -A` has been reduced from 8s to 1.2s.

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by fatedier**
**Please provide a description of this PR:**
Fix #45458

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [X] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by hanxiaop — APPROVED** [review]

File: pilot/pkg/config/kube/crdclient/client.go [hunk 1]
```diff
@@ -80,14 +80,16 @@ type Client struct {
 	logger           *log.Scope
 
 	// namespacesFilter is only used to initiate filtered informer.
-	namespacesFilter func(obj interface{}) bool
+	namespacesFilter    func(obj interface{}) bool
+	fieldSelectorsByGVK map[config.GroupVersionKind]string
```
Comment (hzxuzhonghu) [inline pilot/pkg/config/kube/crdclient/client.go:84]: suggest use map[config.GroupVersionKind]kubetypes.Filter here, so lately we may need other filter condition like label selector

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/crdclient/client.go:84]: I think the use case is by revision

Comment (fatedier) [reply pilot/pkg/config/kube/crdclient/client.go:84]:
```go
filter := kubetypes.Filter{
    ObjectFilter: func(t any) bool {
        if cl.namespacesFilter != nil && !cl.namespacesFilter(t) {
            return false
        }
        return config.LabelsInRevision(t.(controllers.Object).GetLabels(), cl.revision)
    },
}
```

It appears that currently the namespace filter and revision filter are already supported through separate configurations, but only on the client side.

Comment (fatedier) [reply pilot/pkg/config/kube/crdclient/client.go]: updated

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/config/kube/crdclient/client.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/crdclient/client.go]: good

**Pull request merged**